kaldb/src/main/java/com/slack/kaldb/logstore/LuceneIndexStoreImpl.java

```diff
@@ -187,18 +183,20 @@ public void addMessage(LogMessage message) {
       if (indexWriter.isPresent()) {
         indexWriter.get().addDocument(documentBuilder.fromMessage(message));
       } else {
-        LOG.error("IndexWriter should never be null when adding a message");
-        throw new IllegalStateException("IndexWriter should never be null when adding a message");
+        RuntimeException ex =
+            new IllegalStateException("IndexWriter should never be null when adding a message");
+        LOG.error("IndexWriter should never be null when adding a message", ex);
+        throw ex;
       }
     } catch (PropertyTypeMismatchException
         | UnSupportedPropertyTypeException
         | IllegalArgumentException e) {
-      LOG.error(String.format("Indexing message %s failed with error:", message), e);
+      LOG.error(String.format(Locale.ROOT, "Indexing message %s failed with error:", message), e);
       messagesFailedCounter.increment();
     } catch (IOException e) {
       // TODO: For now crash the program on IOException since it is likely a serious issue.
       // In future may need to handle this case more gracefully.
-      e.printStackTrace();
+      LOG.error("Error adding document ", e);
```

Comment:
TODO should be updated to match the error.

Reply:
sorry I don't follow? What error should this match?